lib/MusicBrainz/Server/Entity/MediumCDTOC.pm

```diff
@@ -34,7 +34,7 @@ sub is_perfect_match {
     my @medium_tracks = @{ $self->medium->cdtoc_tracks };
 
     return (@cdtoc_info == @medium_tracks) && all {
-      $_->[0]{length_time} == $_->[1]->length
+      ($_->[0]{length_time} // -1) == ($_->[1]->length // -1)
```

Comment:
Should it be a perfect match if there's no length_time on both sides? Is that even a possible situation in the first place?

Reply:
That was stupid to replace `undef` with `-1` here since `undef` compares to `0` already. It seems more sensible too. Fixed with 095bedfdaa.